From 837065d4ed134dd61f7a384fe19ba1d1b42adaaa Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Fri, 15 Dec 2017 17:58:49 +0900
Subject: [PATCH 106/909] clk: renesas: r8a77965: Replace PLL3 multiplication
 setting

commit 0a2ca3eafa6376af2b18504105602ad999e57ce5 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This patch replaces PLL3 multiplication setting for DDR clock frequency.

  - After changes, new PLL3 multiplication setting:
    MD19 MD17 : DDR clock frequency
Signed-off-by: Meng Li <Meng.Li@windriver.com>
    -------------------------------
      0    0  : DDR3200
      0    1  : DDR2800
      1    0  : DDR2400
      1    1  : DDR1600

  - Before changes, old PLL3 multiplication setting:
    MD19 MD17 : DDR clock frequency
    -------------------------------
      0    0  : DDR3200
      0    1  : DDR2133
      1    0  : Prohibited setting
      1    1  : DDR1600

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
---
 drivers/clk/renesas/r8a77965-cpg-mssr.c | 32 ++++++++++++-------------
 1 file changed, 16 insertions(+), 16 deletions(-)

diff --git a/drivers/clk/renesas/r8a77965-cpg-mssr.c b/drivers/clk/renesas/r8a77965-cpg-mssr.c
index 0e9c0e754532..9e5a54193964 100644
--- a/drivers/clk/renesas/r8a77965-cpg-mssr.c
+++ b/drivers/clk/renesas/r8a77965-cpg-mssr.c
@@ -253,20 +253,20 @@ static const struct mssr_mod_clk r8a77965_mod_clks[] __initconst = {
  * 14 13 19 17	(MHz)
  *-----------------------------------------------------------------
  * 0  0  0  0	16.66 x 1	x180	x192	x192	x144	/16
- * 0  0  0  1	16.66 x 1	x180	x192	x128	x144	/16
- * 0  0  1  0	Prohibited setting
+ * 0  0  0  1	16.66 x 1	x180	x192	x168	x144	/16
+ * 0  0  1  0	16.66 x 1	x180	x192	x144	x144	/16
  * 0  0  1  1	16.66 x 1	x180	x192	x192	x144	/16
  * 0  1  0  0	20    x 1	x150	x160	x160	x120	/19
- * 0  1  0  1	20    x 1	x150	x160	x106	x120	/19
- * 0  1  1  0	Prohibited setting
+ * 0  1  0  1	20    x 1	x150	x160	x140	x120	/19
+ * 0  1  1  0	20    x 1	x150	x160	x120	x120	/19
  * 0  1  1  1	20    x 1	x150	x160	x160	x120	/19
  * 1  0  0  0	25    x 1	x120	x128	x128	x96	/24
- * 1  0  0  1	25    x 1	x120	x128	x84	x96	/24
- * 1  0  1  0	Prohibited setting
+ * 1  0  0  1	25    x 1	x120	x128	x112	x96	/24
+ * 1  0  1  0	25    x 1	x120	x128	x96	x96	/24
  * 1  0  1  1	25    x 1	x120	x128	x128	x96	/24
  * 1  1  0  0	33.33 / 2	x180	x192	x192	x144	/32
- * 1  1  0  1	33.33 / 2	x180	x192	x128	x144	/32
- * 1  1  1  0	Prohibited setting
+ * 1  1  0  1	33.33 / 2	x180	x192	x168	x144	/32
+ * 1  1  1  0	33.33 / 2	x180	x192	x144	x144	/32
  * 1  1  1  1	33.33 / 2	x180	x192	x192	x144	/32
  */
 #define CPG_PLL_CONFIG_INDEX(md)	((((md) & BIT(14)) >> 11) | \
@@ -277,20 +277,20 @@ static const struct mssr_mod_clk r8a77965_mod_clks[] __initconst = {
 static const struct rcar_gen3_cpg_pll_config cpg_pll_configs[16] __initconst = {
 	/* EXTAL div	PLL1 mult/div	PLL3 mult/div	OSC prediv */
 	{ 1,		192,	1,	192,	1,	16,	},
-	{ 1,		192,	1,	128,	1,	16,	},
-	{ 0, /* Prohibited setting */				},
+	{ 1,		192,	1,	168,	1,	16,	},
+	{ 1,		192,	1,	144,	1,	16,	},
 	{ 1,		192,	1,	192,	1,	16,	},
 	{ 1,		160,	1,	160,	1,	19,	},
-	{ 1,		160,	1,	106,	1,	19,	},
-	{ 0, /* Prohibited setting */				},
+	{ 1,		160,	1,	140,	1,	19,	},
+	{ 1,		160,	1,	120,	1,	19,	},
 	{ 1,		160,	1,	160,	1,	19,	},
 	{ 1,		128,	1,	128,	1,	24,	},
-	{ 1,		128,	1,	84,	1,	24,	},
-	{ 0, /* Prohibited setting */				},
+	{ 1,		128,	1,	112,	1,	24,	},
+	{ 1,		128,	1,	96,	1,	24,	},
 	{ 1,		128,	1,	128,	1,	24,	},
 	{ 2,		192,	1,	192,	1,	32,	},
-	{ 2,		192,	1,	128,	1,	32,	},
-	{ 0, /* Prohibited setting */				},
+	{ 2,		192,	1,	168,	1,	32,	},
+	{ 2,		192,	1,	144,	1,	32,	},
 	{ 2,		192,	1,	192,	1,	32,	},
 };
 
-- 
2.17.1

